vcd/resource_vcd_vapp_vm.go

```diff
@@ -961,28 +1062,23 @@ func resourceVcdVAppVmUpdateExecute(d *schema.ResourceData, meta interface{}) er
 			}
 		}
 
-		if d.HasChange("memory") {
-
-			task, err := vm.ChangeMemorySize(d.Get("memory").(int))
+		if memoryNeedsColdChange || executionType == "create" {
+			err = changeMemorySize(d, vm)
 			if err != nil {
-				return fmt.Errorf("error changing memory size: %s", err)
+				return err
 			}
+		}
 
-			err = task.WaitTaskCompletion()
+		if cpusNeedsColdChange || executionType == "create" {
+			err = changeCpuCount(d, vm)
 			if err != nil {
 				return err
 			}
 		}
 
-		if d.HasChange("cpus") || d.HasChange("cpu_cores") {
-			var task govcd.Task
-			var err error
-			if d.Get("cpu_cores") != nil {
-				coreCounts := d.Get("cpu_cores").(int)
-				task, err = vm.ChangeCPUCountWithCore(d.Get("cpus").(int), &coreCounts)
-			} else {
-				task, err = vm.ChangeCPUCount(d.Get("cpus").(int))
-			}
+		if d.HasChange("cpu_cores") {
+			coreCounts := d.Get("cpu_cores").(int)
+			task, err := vm.ChangeCPUCountWithCore(d.Get("cpus").(int), &coreCounts)
```

Comment:
Curious, why was this logic removed?

```
			} else {
				task, err = vm.ChangeCPUCount(d.Get("cpus").(int))
			}
```

Reply:
replaced by 
```
		if cpusNeedsColdChange || executionType == "create" {
			err = changeCpuCount(d, vm)
			if err != nil {
				return err
			}
		}
```

Reply:
But now it seems the code is updating CPUs **twice** if cores has changed. First in line 1076 and second in line 1084.

Also, does this mean that CPUs need VM reboot, but CPU cores do not? Or is it only that you don't have an analogous variable `cpuCoresNeedColdChange`?